bindings/src/main/java/io/pravega/storage/azure/AzureChunkStorage.java

```diff
@@ -0,0 +1,210 @@
+                    }
+                }
+                return retValue;
+            }
+        } catch (IllegalArgumentException e) {
+                throw e;
+        } catch (Exception e) {
+            throw convertException(handle.getChunkName(), "doRead", e);
+        }
+    }
+
+    @Override
+    protected int doWrite(ChunkHandle handle, long offset, int length, InputStream data) throws ChunkStorageException {
+        try {
+            val objectPath = getObjectPath(handle.getChunkName());
+            val metadata = client.getBlobProperties(objectPath);
+            if (metadata.getBlobSize() != offset) {
+                throw new InvalidOffsetException(handle.getChunkName(), metadata.getBlobSize(), offset, "doWrite");
+            }
+            client.appendBlock(objectPath, offset, length, new BufferedInputStream(data));
+            return length;
+        } catch (Exception e) {
+            throw convertException(handle.getChunkName(), "doWrite", e);
+        }
+    }
+
+    @Override
+    protected int doConcat(ConcatArgument[] chunks) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException("AzureChunkStorage does not support concat operation.");
+    }
+
+    @Override
+    protected void doSetReadOnly(ChunkHandle handle, boolean isReadOnly) throws UnsupportedOperationException, ChunkStorageException {
+        throw new UnsupportedOperationException("AzureChunkStorage does not support setReadOnly operation.");
+    }
+
+    @Override
+    @SneakyThrows
+    public void close() {
+        if (shouldClose && !this.closed.getAndSet(true)) {
+            this.client.close();
+        }
+        super.close();
+    }
+
+    private ChunkStorageException convertException(String chunkName, String message, Exception e) {
+        ChunkStorageException retValue = null;
+        if (e instanceof ChunkStorageException) {
+            return (ChunkStorageException) e;
+        }
+        if (e instanceof BlobStorageException) {
+            BlobStorageException blobStorageException = (BlobStorageException) e;
+            val errorCode = blobStorageException.getErrorCode();
+
+            if (errorCode.equals(BlobErrorCode.BLOB_NOT_FOUND)) {
+                retValue = new ChunkNotFoundException(chunkName, message, e);
+            }
+
+            if (errorCode.equals(BlobErrorCode.BLOB_ALREADY_EXISTS)) {
+                retValue = new ChunkAlreadyExistsException(chunkName, message, e);
+            }
+
+            if (errorCode.equals(BlobErrorCode.AUTHENTICATION_FAILED)) {
+                retValue =  new ChunkStorageException(chunkName, String.format("Authentication failed for chunk %s - %s.", chunkName, message), e);
+            }
+        }
+
+        if (retValue == null) {
+            retValue = new ChunkStorageException(chunkName, message, e);
+        }
+
+        return retValue;
+    }
+
+    private String getObjectPath(String objectName) {
```

Comment:
Please add description here also

Reply:
Again, getObjectPath is private, limited to that class and self-explanatory as it returns the ObjectPath itself by pre-pending prefix of the Pravega owned Azure path under the assigned container.